Assets/Scripts/Soul.cs

```diff
@@ -0,0 +1,57 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Soul : MonoBehaviour
+{
+    [SerializeField] private soulType type;
+    private Rigidbody2D soulRigidbody;
+    private bool pickable = false;
+    public soulType Type
+    {
+        get
+        {
+            return type;
+        }
+    }
+    private void OnTriggerEnter2D(Collider2D collision)
+    {
+        if (collision.tag == "Player" && !collision.isTrigger && pickable)
+        {
+            switch (Type)
+            {
+                case Soul.soulType.small:
+                    Debug.Log("Opa chirik");
+                    collision.GetComponent<SoulsController>().Replenishment(10);
+                    Destroy(gameObject);
```

Comment:
You are destroying gameObject in any case, maybe you should take this logic out of the switch